students_proj/src/main/java/com/musala/simple/students/Main.java

```diff
@@ -0,0 +1,172 @@
+package com.musala.simple.students;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import com.google.gson.Gson;
+
+public class Main {
+
+	public static void main(String[] args) {
+
+		String path;
+		
+		// Check if any arguments are provided
+		if (args.length == 0) {
+			throw new IllegalArgumentException("Path is missing.");
+		}
+		
+		path = args[0];
+    	File file = new File(path);
+
+    	// Check if path exists
+    	if (!file.exists()) {
+    		throw new IllegalArgumentException("Invalid path specified.");
+    	}
+    		
+    	// Check if the valid path is a file
+	    if (!file.isFile()) {
+	    	throw new IllegalArgumentException("The path specified is not an existing file.");
+	    }	
+	    
+        // Check the file extension to be JSON.
+	    boolean isJson = path.endsWith(".json");
+        if (!isJson) {
```

Comment:
Since the expression `path.endsWith(".json")` is self-explanatory enough, there is no need to have the `isJson` variable. Furthermore, if you have already defined a constant for the json extension, the row can simply take the form: `if (!path.endsWith(FILE_EXTENSION_JSON))`.